### PR TITLE
Fix/mnt 23087 export of records failing

### PR DESCRIPTION
**Pull request by antoniojfelix**

### CODE REVIEW

File: repository/src/main/java/org/alfresco/repo/exporter/ExporterComponent.java [hunk 1]
```diff
@@ -1,28 +1,28 @@
-/*
- * #%L
- * Alfresco Repository
- * %%
- * Copyright (C) 2005 - 2016 Alfresco Software Limited
```
Comment (jottley) [inline repository/src/main/java/org/alfresco/repo/exporter/ExporterComponent.java:5]: Let's take the opportunity to update the copyright dates

File: repository/src/main/java/org/alfresco/repo/exporter/ExporterComponent.java [hunk 2]
```diff
@@ -1022,24 +1065,80 @@ public ExporterContextImpl(ExporterCrawlerParameters parameters)
...
+            }
+            else {
+                return index < exportList.length;
+            }
         }
         
         public int setNextValue()
         {
-            return ++index;
+            if(exportListMap!=null && index==exportListMap.get(indexSubList).length-1){
```
Comment (jottley) [inline repository/src/main/java/org/alfresco/repo/exporter/ExporterComponent.java:1131]: Let's make sure we get this formatted correctly

File: repository/src/main/java/org/alfresco/repo/exporter/ViewXMLExporter.java [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
- * 
- * You should have received a copy of the GNU Lesser General Public License
- * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
- * #L%
- */
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2016 Alfresco Software Limited
```
Comment (jottley) [inline repository/src/main/java/org/alfresco/repo/exporter/ViewXMLExporter.java:5]: Copyright date update

File: repository/src/main/java/org/alfresco/service/cmr/view/ExporterService.java [hunk 4]
```diff
@@ -1,28 +1,28 @@
...
- * 
- * You should have received a copy of the GNU Lesser General Public License
- * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
- * #L%
- */
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2016 Alfresco Software Limited
```
Comment (jottley) [inline repository/src/main/java/org/alfresco/service/cmr/view/ExporterService.java:5]: Update copyright date

File: repository/src/test/java/org/alfresco/repo/exporter/ExporterComponentTest.java [hunk 5]
```diff
@@ -153,15 +156,62 @@ public void testExport()
...
+        TestProgress testProgress = new TestProgress();
+        Location location = new Location(storeRef);
+
+        // now export
+        location.setPath("/system");
+        File tempFile = TempFileProvider.createTempFile("xmlexporttest", ".xml");
+        OutputStream output = new FileOutputStream(tempFile);
+        ExporterCrawlerParameters parameters = new ExporterCrawlerParameters();
+        parameters.setExportFrom(location);
+        //        parameters.setExcludeAspects(new QName[] { ContentModel.ASPECT_AUDITABLE });
```
Comment (jottley) [inline repository/src/test/java/org/alfresco/repo/exporter/ExporterComponentTest.java:183]: If we don't need them let's remove the commented out code

File: repository/src/test/java/org/alfresco/repo/exporter/ExporterComponentTest.java [hunk 6]
```diff
@@ -153,15 +156,62 @@ public void testExport()
         parameters.setExportFrom(location);
 //        parameters.setExcludeAspects(new QName[] { ContentModel.ASPECT_AUDITABLE });
 //        parameters.setExcludeChildAssocs(new QName[] { ContentModel.ASSOC_CONTAINS });
-        
+
```
Comment (jottley) [inline repository/src/test/java/org/alfresco/repo/exporter/ExporterComponentTest.java:159]: Do we have any checks to verify that it is working as expected now that you are using a new approach?

**Review by jottley — APPROVED** [review]

**Review by suneet-gupta — APPROVED** [review]
LGTM

**Pull request merged**